src/components/ThemeProviderWrapper.tsx

```diff
@@ -0,0 +1,27 @@
+import { createTheme, ThemeProvider } from "@material-ui/core";
+import { ReactElement, useMemo, useState } from "react";
+
+import { ThemeContext } from "../contexts/ThemeContext";
+import { themes } from "../types/ThemeTypes";
+
+export default function ThemeProviderWrapper(props: {
+  children: ReactElement;
+}) {
+  const [currentTheme, setCurrentTheme] = useState<themes>(themes.light);
+
+  const theme = useMemo(
+    () =>
+      createTheme({
+        palette: {
+          type: currentTheme,
+        },
```

Comment:
Are themes, both dark and light, considered properly set up at this point?
Because it seems to be necessary to have a working themes after that one.

Reply:
dark and light theme works fine, if we want themes with unique primary/secondary colors we can add them